### DISCUSSION TITLE
The Whisky Application: Is it possible to update Wine, to the latest version?

### DISCUSSION THREAD

**Discussion by ferreira-ff** [Whisky-App/Whisky#257] [General]
Hello. Whisky is already very good and functional, at version 1.0.1. Regarding GPTK 0.0.2; well, let us wait and see, what Apple will do next. About Wine; Whisky uses the 7.7.0 version. Is there a way to update it, to the latest stable version?
Would it be possible to upgrade it, to the version WineCX22.1.1-8?

I am also having good results, with WineSkin; (**Wrapper: Wineskin-2.9.1.9 & Engine: WineCX22.1.1-8 + Apple GPTK 0.2**)

Again; thank you for the great work, from everyone involved. Kind regards to all.

**Comment by JoshBashed** [comment]
No, it's not possible as of yet.

Reason: the wine included with GPTK is modified in order to be compatible with dxmetal

**Reply by ferreira-ff** [reply]
I see. Well, thank you for the explanation. Let us wait for Apple to update the GPTK; maybe with the release of "Crossover / Wine 23", we will have nice improvements, both to Whisky and WineSkin.

Kind regards

**Reply by odahcam** [reply]
I'm eager for that as well, as of now I can't play my games :(

**Reply by JoshBashed** [reply]
It is now possible. We just won't do it until crossover (basically a paid version of whisky) moves to wine 9. We want to avoid undercutting them.

**Reply by LeaveNhA** [reply]
No offense, but LOL @JoshuaBrest.

**Reply by JoshBashed** [reply]
> No offense, but LOL @JoshuaBrest.

Hey, if it were up to me, we would've been on Wine 9 ages ago. Unfortunately, even though it technically works, it's not perfect and a lot of apps just don't run. Trying to fix issues is not an easy task and requires lots of time.

If you care so much, go figure your own fork of whiskey and Wine. Just because we're open source doesn't mean our time is worth nothing. Please don't go to other open source softwares and demand features. It's free.

**Comment by Gcenx** [comment]
Wineskin wine builds are highly customized and are _not_ for use by other third-partly projects, PortingKit by PaulTheTall is not considered a third-party project. These became branded to persuade other projects from using them.

Wineskin does support D3DMetal right after the release of `game-porting-toolkit` it simply wasn’t advertised until I was happy with it. Wineskin was also the first project to directly bundle D3DMetal once Apple updated there license.

The version of DXVK shipped with Whisky was taken from my fork.

**Comment by IsaacMarovitz** [comment]
After the release of CrossOver 23.5 included D3DM, I decided not to upgrade to Wine 8, at least not while we're using CrossOver's fork of Wine. This is to prevent the undercutting of CodeWeavers' revenue, as they weren't the most pleased about Whisky using their tech, and making the paid part (the user-friendly app) effectively free.

There are plans to switch from CrossOver's Wine to our own fork of mainline Wine, but this will take time, and it's not immediately on the horizon.

In the meantime, if you want to use D3DM with Wine 8, buy a CrossOver 23 license.

**Reply by Gerzer** [reply]
If CrossOver’s fork is open-source and if you’re complying with the text of its license, then why do they get a say in what you do or don’t build with it?

**Reply by JoshBashed** [reply]
> If CrossOver’s fork is open-source and if you’re complying with the text of its license, then why do they get a say in what you do or don’t build with it?

@Gerzer I can see you are very into open source. The short answer is that they don't, but, Issac has decided that they don't want to compete with Codeweaver's paid software known as crossover. Codeweavers is also a big maintainer of Wine. It seems like Isacc doesn't want to drive them out of business or burn any bridges.

If you want, however, you are free to maintain your own version of Whisky and Wine that uses Wine 8. 😉

**Reply by chaosdrop** [reply]
Any updates on this? Crossover seems to be on v24 now. I tired their paid software and it wasn't as good of an experience as Whisky... So wound up uninstalling it and moving back to Whisky.

**Reply by JoshBashed** [reply]
> Any updates on this? Crossover seems to be on v24 now. I tired their paid software and it wasn't as good of an experience as Whisky... So wound up uninstalling it and moving back to Whisky.

We're planning someday to move to Wine 8 but a lot of software are not compatible out of the box.

**Reply by chaosdrop** [reply]
Seems fair, maybe that's one of the reasons Whisky is better than CrossOver, anyway. XD I just came looking because of always seeing, in the terminal, the warnings to upgrade from 7.7 because it's unsupported.

**Reply by Gcenx** [reply]
It’s not that Whisky is better but CrossOver has too many options that I guess confuse user.

> I just came looking because of always seeing, in the terminal, the warnings to upgrade from 7.7 because it's unsupported.

That’s a warning from winetricks, it’s warning is correct as it only ever guarantees the verbs will work with current upstream wine aka wine-9.0, but there’s currently no verbs I can think of that aren’t still working with wine-7.7 currently that aren’t marked.

**Reply by NickStrupat** [reply]
> > If CrossOver’s fork is open-source and if you’re complying with the text of its license, then why do they get a say in what you do or don’t build with it?
> 
> @Gerzer I can see you are very into open source. The short answer is that they don't, but, Issac has decided that they don't want to compete with Codeweaver's paid software known as crossover. Codeweavers is also a big maintainer of Wine. It seems like Isacc doesn't want to drive them out of business or burn any bridges.
> 
> If you want, however, you are free to maintain your own version of Whisky and Wine that uses Wine 8. 😉

Realistically, CodeWeavers contributes to Wine because:
1) they sell a product that is built on it, so they're financially incentivized to improve it for their existing and potential customers, and (the important part...)
2) they are legally obligated, by Wine's license, to contribute those changes back into the Wine project.

I've seen a thousand comments that say how great they think CodeWeavers is for shepherding the Wine project. In reality, they have to if they want to improve the product (which wouldn't exist without the real open-source contributors who started Wine).

I'm not saying CodeWeavers is evil or anything. If you zoom out, they're just another software company that sells a product that's built using a bunch of open-source projects (and obviously some hard work). But the seemingly widely held belief that CodeWeavers is just out there fighting the good fight for the lowly mac gamer is (to me) silly.

Maybe I'm wrong about the above, but if I'm right there much less of a reason to avoid using the latest versions of Wine and the open-source parts of CrossOver.

**Reply by JoshBashed** [reply]
> > > If CrossOver’s fork is open-source and if you’re complying with the text of its license, then why do they get a say in what you do or don’t build with it?
> 
> > 
> 
> > @Gerzer I can see you are very into open source. The short answer is that they don't, but, Issac has decided that they don't want to compete with Codeweaver's paid software known as crossover. Codeweavers is also a big maintainer of Wine. It seems like Isacc doesn't want to drive them out of business or burn any bridges.
> 
> > 
> 
> > If you want, however, you are free to maintain your own version of Whisky and Wine that uses Wine 8. 😉
> 
> 
> 
> Realistically, CodeWeavers contributes to Wine because:
> 
> 1) they sell a product that is built on it, so they're financially incentivized to improve it for their existing and potential customers, and 
> 
> 2) they are legally obligated, by Wine's license, to make those changes directly into the Wine project.
> 
> 
> 
> I've seen a thousand comments that say how great they think CodeWeavers is for shepherding the Wine project. In reality, they have to if they want to improve the product (which wouldn't exist without the real open-source contributors who started Wine).
> 
> 
> 
> I'm not saying CodeWeavers is evil or anything. If you zoom out, they're just another software company that sells a product that's built using a bunch of open-source projects (and obviously some hard work). But the seemingly widely held belief that CodeWeavers is just out there fighting the good fight for the lowly mac gamer is (to me) silly.

Thank you for your input. This is open source software. If you wanna make a fork that uses crossover wine, you are free to do so.

**Reply by NickStrupat** [reply]
Yes, of course I am free to do so. Do you have any thoughts/opinions about what I said? I am genuinely curious if it makes sense and/or affects at all the way you think about CodeWeavers.

**Reply by JoshBashed** [reply]
> Yes, of course I am free to do so. Do you have any thoughts/opinions about what I said? I am genuinely curious if it makes sense and/or affects at all the way you think about CodeWeavers.

From what I know CodeWeavers maintained wine even before it was licensed under GPL. This meant that if they wanted to keep it close sourced, they were able to do so. There was another company that tried to undercut codeweavers by using wine and keeping their changes closed source.

We are free to update to the latest crossover, but it could also put peoples jobs at CodeWeaver's at risk, which would put the development at risk too.

**Reply by Gcenx** [reply]
The _only_ reason wine is still functional on macOS is due to CodeWeavers, without them it would quickly go back to being Linux _only_

Go ahead and build pure upstream wine straight from git (without any patches) and tell me how that works for you.

Even the upstream Winehq macOS packages I provide requires a number of hacks to make them somewhat viable on macOS/Rosetta2, those hacks come from CodeWeavers.

**Reply by SoCuul** [reply]
Is there any way to use Whisky with the latest version of CrossOver 24? I purchased a CrossOver license, but have heard that performance through Whisky using the GPTK could improve the experience. I know Whisky is using an outdated version of wine to not undermine codeweavers revenue, but is there any way for the end-user to use their installation of CrossOver with Whisky?

**Reply by pugsii** [reply]
Bumping this because I'm interested in this too. I hate software with tiered pricing, but CrossOver already earned my money once and my license served me well even years after it had expired. Got a new laptop though, and decided to try out whisky instead. Would love an option to buy a CrossOver license (through a Whisky referral) that would let me upgrade to the latest version available. Could this be a possibility?

**Reply by Gcenx** [reply]
@pugsii Whiskys readme provides an affiliate url

**Reply by Gcenx** [reply]
> Is there any way to use Whisky with the latest version of CrossOver 24? I purchased a CrossOver license, but have heard that performance through Whisky using the GPTK could improve the experience. I know Whisky is using an outdated version of wine to not undermine codeweavers revenue, but is there any way for the end-user to use their installation of CrossOver with Whisky?

@SoCuul Whisky actually runs worse overall the difference is D3DMetal is force enabled along with msync.

There’s no way for Whisky-App to make use of CrossOver wine, that would create a support nightmare for both CodeWeavers & Whisky.

Adding a warning wouldn’t be enough, that was attempted with CXPatcher and _yeah_ I was shows the amount of invalid tickets from CXPatcher users.

**Reply by pugsii** [reply]
> @pugsii Whiskys readme provides an affiliate url

Yes I saw the affiliate link. But I was hoping that a proof of purchase might sway Isaac into supporting the latest version of wine officially in Whisky, without fear of robbing CodeWeaver's revenue. Whisky's UI and ease of use is just a good value add for me.
